core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -97,12 +102,14 @@ public TableMetadata current() {
 
   @Override
   public TableMetadata refresh() {
+    Endpoint.check(endpoints, Endpoint.V1_LOAD_TABLE);
     return updateCurrentMetadata(
         client.get(path, LoadTableResponse.class, headers, ErrorHandlers.tableErrorHandler()));
   }
 
   @Override
   public void commit(TableMetadata base, TableMetadata metadata) {
+    Endpoint.check(endpoints, Endpoint.V1_UPDATE_TABLE);
```

Comment:
Shouldn't this be a `CREATE` or `UPDATE` check (possibly in the switch statement below)?

Reply:
checking for `V1_UPDATE_TABLE` is correct here, because we're only hitting this endpoint here. `UpdateType.CREATE` is only used during staged table creation in https://github.com/apache/iceberg/blob/9ddd6d34bafad2abed19c089d7d8758aa6150268/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L826-L842 and we're already checking against `V1_CREATE_TABLE` there

Reply:
Makes sense, I was thinking more in terms of the operation, not the endpoint.  You're right.